src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -461,58 +465,123 @@ private string GetAzDevStagingDirectory()
         /// <param name="location">Location of asset</param>
         /// <param name="locationType">Type of location</param>
         /// <param name="nonShipping">If true, the asset is not intended for end customers</param>
-        internal void AddAsset(List<AssetData> assets, string assetName, string version, string location, LocationType locationType, bool nonShipping)
+        internal void AddAsset(List<AssetData> assets, string assetName, string version, string location,
+            LocationType locationType, bool nonShipping)
         {
             assets.Add(new AssetData(nonShipping)
             {
-                Locations = (location == null) ? null : ImmutableList.Create(new AssetLocationData(locationType)
-                {
-                    Location = location,
-                }),
+                Locations = (location == null)
+                    ? null
+                    : ImmutableList.Create(new AssetLocationData(locationType)
+                    {
+                        Location = location,
+                    }),
                 Name = assetName,
                 Version = version,
             });
         }
 
-        internal BuildData MergeBuildManifests(List<BuildData> buildsMetadata)
+        internal (List<PackageArtifactModel>,
+            List<BlobArtifactModel>) GetPackagesAndBlobsInfo(Manifest manifest)
+        {
+            List<PackageArtifactModel> packageArtifacts = new List<PackageArtifactModel>();
+            List<BlobArtifactModel> blobArtifacts = new List<BlobArtifactModel>();
+
+            foreach (var package in manifest.Packages)
+            {
+                PackageArtifactModel packageArtifact = new PackageArtifactModel()
+                {
+                    Attributes = new Dictionary<string, string>
+                    {
+                        { "NonShipping", package.NonShipping.ToString().ToLower() },
+                    },
+                    Id = package.Id,
+                    Version = package.Version
+                };
+
+                packageArtifacts.Add(packageArtifact);
+            }
+
+            foreach (var blob in manifest.Blobs)
+            {
+                BlobArtifactModel blobArtifact = new BlobArtifactModel()
+                {
+                    Attributes = new Dictionary<string, string>
+                    {
+                        { "NonShipping", blob.NonShipping.ToString().ToLower() },
+                        {
+                            "Category",
+                            !string.IsNullOrEmpty(blob.Category) ? blob.Category.ToString().ToUpper() : NoCategory
+                        }
+                    },
+                    Id = blob.Id,
+                };
+
+                blobArtifacts.Add(blobArtifact);
+            }
+
+            return (packageArtifacts, blobArtifacts);
+
+        }
+
+        internal Manifest CheckIfManifestCanBeMerged(List<Manifest> parsedManifest)
```

Comment:
[nit] I think I have this feedback earlier as well, but since this is a list, it should be plural. parsedManifests

Reply:
done